dc/core/PixelStreamUpdater.cpp

```diff
@@ -153,25 +156,42 @@ QRect toRect( const deflect::SegmentParameters& params )
     return QRect( params.x, params.y, params.width, params.height );
 }
 
+void PixelStreamUpdater::_computeVisibleSet( const deflect::Segments& segments )
+{
+    _visibleSet.clear();
```

Comment:
Not possible, because the _visibleSet must be adjusted to the new list of segments.
We "could" early out if the number of segments is the same, but there is no guarantees that the segment sizes don't change, so I'm not sure if it is a wise idea...